js/interface.js

```diff
@@ -114,21 +130,49 @@ var dsQueryProvider = Fliplet.Widget.open('com.fliplet.data-source-query', {
 });
 
 dsQueryProvider.then(function onForwardDsQueryProvider(result) {
-  var validation = {
-    sms: {
-      toColumn: result.data.columns.smsTo,
-      matchColumn: result.data.columns.smsMatch,
-      template: $('#sms-template').val(),
-      expire: $('#expire-timeout').val(),
-    },
-    email: {
-      toColumn: result.data.columns.emailMatch,
-      matchColumn: result.data.columns.emailMatch,
-      template: emailProviderResult || defaultEmailSettings,
-      expire: $('#expire-timeout').val(),
-    }
+  // If there is no data source just save it althoug not valid config
+  if (!dataSource) {
+    Fliplet.Widget.save();
   }
 
+  var validation = dataSource.definition.validation || {};
+
+  // Let's update only the selected verification type on the datasurce settings
+  switch(data.type) {
+    case 'sms':
+      validation.sms = {
+        toColumn: result.data.columns.smsTo,
+        matchColumn: result.data.columns.smsMatch,
+        template: $('#sms-template').val(),
+        expire: $('#expire-timeout').val(),
+      };
+      break;
+
+    case 'email':
+      validation.email = {
+        toColumn: result.data.columns.emailMatch,
+        matchColumn: result.data.columns.emailMatch,
+        template: emailProviderResult || defaultEmailSettings,
+        expire: $('#expire-timeout').val(),
+      }
+      break;
+
+    case 'domain':
```

Comment:
You wouldn't get a `domain` type validation. Domain whitelisting is a `email` type validation with domain whitelisting enabled as an additional feature.

e.g. Search everything in the datasource for the user's email, and also allow it if it it's in one of these domains.

**We should resolve this later though. I don't want to delay the launch of SMS verification by even a minute.**